lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -25,83 +25,105 @@ def __init__(self, context):
     def execute(self) -> int:
         errors = 0
 
-        if self.verify_tool_group(self.context.group) != 0:
-            return 1
-
-        try:
-            remotes = self.context.remote.split(",")
-        except Exception:
-            # We were not given any remotes on the command line, build the list
-            # from the tools group directory.
-            remotes = self.remote(self.tool_group_dir)
-            if not remotes:
-                # FIXME:  This is a weird case -- the group directory exists,
-                #  but contains no remotes. This is the result of clearing the
-                #  tools from a group which has already been cleared, because
-                #  the first clear does not remove the group directory.  This
-                #  behavior might be desirable for the `default` directory, but
-                #  it seems wrong for custom groups.
-                self.logger.error(f'No such group "{self.context.group}".')
-                return 1
-
-        tool_not_found = True  # Not found, yet
-        for remote in remotes:
-            tg_dir_r = self.tool_group_dir / remote
-            if not tg_dir_r.exists():
-                self.logger.warn(
-                    'No remote host "%s" in group %s.', remote, self.context.group,
-                )
+        groups = self.context.group.split(",")
+        for group in groups:
+            if self.verify_tool_group(group) != 0:
+                self.logger.error(f'No such group "{group}".')
```

Comment:
Hmm; this weird pattern of an f-string inside a `Logger.error` is already in use in the file, but if we're using the `Logger` we should be using the standard substitution parameter format: `self.logger.error("No such group %s", group)`.

Although it's bad enough we're using `Logger` for user feedback that perhaps this minor oddness doesn't even register. 😆 

Furthermore, the old code failed immediately on a bad group, so the `Logger.error` made (some) sense; since we're *not* failing now, it should probably be downgraded to `Logger.warn`. (Not that it matters since the `Logger` is configured to emit only the message, anyway. 🙄 )

Reply:
Yeah I agree it should be of level `warn` now

Reply:
I suspect that using `%s` instead of an f-string amounts to six of one vs. a half-dozen of the other.  Either way, here, the group needs to be inserted into the output message, and I'm not sure it matters whether it's done by the caller or by the callee.  I think it comes down to reader preference and cultural compatibility (and I can't help thinking that `%s` looks more like C than Python.  😁)

I'll concur that using `Logger` to produce diagnostics for the end-user, does seem a bit weird.  However, after your-two complaints about the Tool Meister logging, and the fact that the `Logger` is easily manipulated by unit testing, makes me think that perhaps this is being crazy like a fox:  in the default case, it goes to `stderr`, which is where we want it; in the other cases, we can easily send it somewhere better; so...win-win?

And, yeah, if we're not failing, then `warn` would be much more appropriate, I think.

Reply:
Well, there's a "logger style" which defers formatting until the message is emitted vs this which formats on the call. It's a trivial distinction here because we don't expect `warning` or `error` to be disabled, but it still violated the expected style; and, especially the *mix* of those two styles in the same module is distracting. ☹️

Reply:
I suppose that deferring formatting could be quite useful, as long as the overhead in receiving and storing the values is less than the cost of having the caller do the formatting.

And, yes, I'm a fan of consistency...if only we could decide whether to use f-strings, or `%s` specifiers, or `{}` replacers, or....  🙄

Reply:
Yes, the fact that the agent code uses the default `Logger` stdio `%s` style formatting while the server code changes the formatter to the more "modern-feeling" `.format` style `{}` doesn't help things when we're moving back and forth. Mixing it with f-strings and sometimes `.format()` calls as well (sometimes in the same module) makes it even worse.

And again, this whole thing is messier because these are really user feedback messages *not* the "error logs" for which `Logger` is really intended. I'd much rather see all of these as `click.echo(... , err=True)`, and f-strings would be just fine since it's inherently obvious we need the formatting whenever we hit the statement.